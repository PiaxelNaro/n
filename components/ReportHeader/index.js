// @flow
import React from 'react';
import styled from 'styled-components';

const MainTitle = styled.h1`
    font-size: 26px;
    color: #666;
    font-weight: normal;
    padding: 0;
    margin: 5px 0;
`;

const SubTitle = styled.h3`
    color: #888;
    font-size: 15px;
    font-weight: normal;
    padding: 0;
    margin: 0;
    text-transform: uppercase;
`;

const TitleLink = styled.a`
    color: #5fafd7;
    margin-left: 5px;
    text-decoration: none;
    outline-style: none;
    cursor: pointer;
    font-size: 13px;

    &:hover, &:active {
        text-decoration: underline;
    }
`;

const ReportHeader = ({
    title,
    subtitle,
    link,
}: {
    title: string,
    subtitle?: string,
    link: Object,
}) => (
    <div>
        <SubTitle>{subtitle}</SubTitle>
        <MainTitle>
            {title}
            <TitleLink target="_blank" rel="noopener noreferrer" href={link.url}>
                {link.text}
            </TitleLink>
        </MainTitle>
    </div>
);

export default ReportHeader;
