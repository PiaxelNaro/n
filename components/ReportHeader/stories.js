import React from 'react';
import { storiesOf } from '@kadira/storybook';
import ReportHeader from '.';

storiesOf('ReportHeader', module)
    .add('Dashboard Report Header', () => (
        <ReportHeader
            subtitle="subsub title"
            title="Titletitle"
            link={{
                text: 'About this report',
                url: 'https://www.google.com',
            }}
        />
    ))
    .add('Report Header Without Subtitle', () => (
        <ReportHeader
            title="Titletitle"
            link={{
                text: 'About this report',
                url: 'https://www.google.com',
            }}
        />
    ));
