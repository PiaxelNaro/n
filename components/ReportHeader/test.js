import React from 'react';
import { shallow } from 'enzyme';

import ReportHeader from '.';

describe('<ReportHeader>', () => {
    it('Render UI', () => {
        const reportHeader = shallow(
            <ReportHeader
                subtitle="SUBSUB TITLE"
                title="Titletitle"
                link={{
                    text: 'About this report',
                    url: 'https://www.google.com',
                }}
            />
        );
        expect(reportHeader).toMatchSnapshot();
    });
});
