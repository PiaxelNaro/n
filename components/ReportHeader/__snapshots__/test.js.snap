// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportHeader> Render UI 1`] = `
<div>
  <ReportHeader__SubTitle>
    SUBSUB TITLE
  </ReportHeader__SubTitle>
  <ReportHeader__MainTitle>
    Titletitle
    <ReportHeader__TitleLink
      href="https://www.google.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      About this report
    </ReportHeader__TitleLink>
  </ReportHeader__MainTitle>
</div>
`;
